lib/src/oauth2.dart

```diff
@@ -120,18 +122,26 @@ Future<T> withClient<T>(SystemCache cache, Future<T> Function(Client) fn) {
 ///
 /// If saved credentials are available, those are used; otherwise, the user is
 /// prompted to authorize the pub client.
-Future<Client> _getClient(SystemCache cache) async {
-  var credentials = _loadCredentials(cache);
-  if (credentials == null) return await _authorize();
-
-  var client = Client(credentials,
-      identifier: _identifier,
-      secret: _secret,
-      // Google's OAuth2 API doesn't support basic auth.
-      basicAuth: false,
-      httpClient: httpClient);
-  _saveCredentials(cache, client.credentials);
-  return client;
+Future<Client> _getClient(SystemCache cache, Uri requestServer) async {
+  var isThirdPartyServer =
+      !runningFromTest && requestServer.host != 'https://pub.dartlang.org';
```

Comment:
@stefandevo: please extract the hard-coded hostname (and URLs) into a single place and reference only the extracted variables. It will help to find all the references that are using the same hostname.

Reply:
Ok will do; note that in the codebase there are hardcoded references already to this hostname, hence why I reused it like that. Ok that I replace it in the whole codebase? (I did get a remark before that I should limit the PR to the scope)

Reply:
If @jonasfj is OK with this, let's keep it this way. We can do a follow-up PR that extracts them.